app/views/services/_connection_menu.html.erb

```diff
@@ -0,0 +1,111 @@
+                data-component-type="textarea"><a href="#add-page"><%= t('components.list.textarea') -%></a></li>
+            <li data-page-type="singlequestion"
+                data-component-type="email"><a href="#add-page"><%= t('components.list.email') -%></a></li>
+            <li data-page-type="singlequestion"
+                data-component-type="number"><a href="#add-page"><%= t('components.list.number') -%></a></li>
+            <li data-page-type="singlequestion"
+                data-component-type="date"><a href="#add-page"><%= t('components.list.date') -%></a></li>
+            <li data-page-type="singlequestion"
+                data-component-type="radios"><a href="#add-page"><%= t('components.list.radios') -%></a></li>
+            <li data-page-type="singlequestion"
+                data-component-type="checkboxes"><a href="#add-page"><%= t('components.list.checkboxes') -%></a></li>
+            <li data-page-type="singlequestion"
+                data-component-type="upload"><a href="#add-page"><%= t('components.list.upload') -%></a></li>
+          </ul>
+        </li>
+        
+        <li data-page-type="multiplequestions"><a href="#add-page"><%= t('actions.add_multi_question') -%></a></li>
+        
+        <li data-page-type="content"><a href="#add-page"><%= t('actions.add_content') -%></a></li>
+
+        <li data-page-type="exit"><a href="#add-page"><%= t('actions.add_exit') -%></a></li>
+    <% end %>
+
+    <% if !service.checkanswers_page.present? && item[:type] != 'flow.branch' && item[:next].blank?  %>
+      <li data-page-type="checkanswers"><a href="#add-page"><%= t('actions.add_check_answers') -%></a></li>
+    <% end %>
+      
+    <% if item[:type] == 'page.checkanswers' %>
+      <% if service.confirmation_page.present? %>
+        <% if !@publish_warning&.confirmation_in_main_flow? %>
+          <li data-action="reconnect-confirmation"
+              data-url="<%= api_service_flow_destinations_path(service.service_id, item[:uuid]) -%>"
+              data-destination-uuid="<%= service.confirmation_page.uuid -%>">
+              <a href="#reconnect-confirmation"><%= t('actions.reconnect_confirmation') -%></a>
+          </li>
+        <% end %>
+      <% else %>
+        <li data-page-type="confirmation"><a href="#add-page"><%= t('actions.add_confirmation') -%></a></li>
+      <% end %>
+    <% end %>
+
+    <% unless (item[:type] =~ /page.(checkanswers|confirmation|exit)/) || item[:type] == 'flow.branch' %>
+      <% unless item[:type] == 'page.start' %>
+        <li data-action="link">
+          <%= link_to t('services.branch'),
+          new_branches_path(service.service_id, item[:uuid]) %>
+        </li>
+      <% end %>
+
+      <li data-action="destination">
+        <%= link_to t('actions.change_destination'),
+        new_api_service_flow_destination_path(service.service_id, item[:uuid]) %>
+      </li>
+    <% end %>
+
+  <% else %>
+
+    <% unless item[:type] == 'page.checkanswers' %>
```

Comment:
At a glance, it looks like we could reduce some duplication here but I wouldn't let that stand in the way of getting this merged. We can deal with idealistic coding at a later date, if need be.